src/address.rs

```diff
@@ -3,36 +3,53 @@ use serde::Serializer;
 use std::str;
 use std::str::FromStr;
 use utils::{hex_str_to_bytes, ByteDecodeError};
-
 /// This type represents ETH address
 #[derive(PartialEq, Debug, Clone)]
-pub struct Address([u8; 20]);
+pub struct Address {
```

Comment:
does eth types have value as a library? hopefully it would have more validation on the address properties.

Reply:
Actually there is little to none validation of the Address on the address type itself. Maybe just that it should stay below 20 bytes.

Reply:
Verification of length of the key (when parsing from a string) is already covered.